pkg/local_object_storage/blobstor/fstree/fstree.go

```diff
@@ -137,11 +137,19 @@ func (t *FSTree) iterate(depth uint64, curPath []string, prm common.IteratePrm)
 
 		if prm.LazyHandler != nil {
 			err = prm.LazyHandler(*addr, func() ([]byte, error) {
-				return os.ReadFile(filepath.Join(curPath...))
+				data, err := os.ReadFile(filepath.Join(curPath...))
+				if err != nil && os.IsNotExist(err) {
+					return nil, logicerr.Wrap(apistatus.ObjectNotFound{})
```

Comment:
I'm really not sure this `ObjectNotFound` wrapping is good. What other `LazyHandler`s we have, can we have another source of `apistatus.ObjectNotFound` that won't be related to files missing between listing and action?

Reply:
currently all callers just skip such cases, so IMO it'd be easier to skip them here

Reply:
> I'm really not sure this ObjectNotFound wrapping is good.

@roman-khimov, why not? that is the idea of the `logicerr` (at least how i remembered it): every err that can be caused by non-disk errors should be _logic_ (and that line will be executed in the iterate-remove-read scenarios most likely). if drop it will be easy to make a bug and increment the internal disk errors counter and move a shard to the degraded mode

> What other LazyHandlers we have

WC only (blobovnicza can be called with that func too but it caches objects so it is impossible to `stat`+`read`+"oopps")

> can we have another source of apistatus.ObjectNotFound that won't be related to files missing between listing and action?

`Get`/etc only. not in the `Iterate`

Reply:
To me `apistatus.ObjectNotFound` is something that we return for networked requests, not something that we can have locally. At the same time if we're to ever have any other non-FS backends, there may be no `fs.ErrNotExist`, something else would be returned and then we'll have to unify them with some common error anyway.

Reply:
> To me apistatus.ObjectNotFound is something that we return for networked requests, not something that we can have locally.

i do agree that it can be revised but now that is what we have: SE returns `apistatus` errors. do not have a strict opinion about that RN